package/piwebcam/multi-gadget.sh

```diff
@@ -64,17 +66,13 @@ EOF
 config_usb_webcam () {
   mkdir -p functions/uvc.usb0/control/header/h
 
-  config_frame mjpeg m  640  360
-  config_frame mjpeg m  640  480
-  config_frame mjpeg m  800  600
-  config_frame mjpeg m 1024  768
-  config_frame mjpeg m 1280  720
-  config_frame mjpeg m 1280  960
-  config_frame mjpeg m 1440 1080
-  config_frame mjpeg m 1536  864
-  config_frame mjpeg m 1600  900
-  config_frame mjpeg m 1600 1200
-  config_frame mjpeg m 1920 1080
+  grep -E "(mjpeg|raw|mp4)\s+[[:digit:]]+\s+[[:digit:]]+" $VIDEO_FORMATS_FILE | sed -E "s/\s+/ /g" | while read -r line
+  do
+    VIDEO_FORMAT=$(echo "$line" | cut -d ' ' -f1)
+    X=$(echo "$line" | cut -d ' ' -f2)
+    Y=$(echo "$line" | cut -d ' ' -f3)
+    config_frame "$VIDEO_FORMAT" m "$X" "$Y"
```

Comment:
"m" should be the first character of the format instead of being hardcoded

Reply:
To be honest I do not know what the "m" parameter in the config file function call does. It creates some file for USB configuration. So does it have to be changeable? How should the config file look like?

    m mjpeg 1920 1080

or

    mmjpeg 1920 1080

In the second case I would just take the very first character. What about the syntax checking? Should it be only "m" at the moment?

Reply:
That single character is the specific streaming header descriptor, "m" for mjpeg and "u" for uncompressed. I *think* you could use any letter, hence I've said to just use the first letter of the format.